abbyy_to_epub3/create_epub.py

```diff
@@ -314,33 +331,46 @@ def extract_images(self):
                 cover_leaf = p
                 break
         try:
-            cover_leaf
+            return cover_leaf
         except NameError:
             e = "No pages in scandata marked as Cover, Title, or Normal"
             self.logger.error(e)
             raise RuntimeError(e)
 
+    def extract_cover(self):
+        """
+        http://web.archive.org/web/20180416230000/https://www.safaribooksonline.com/blog/2009/11/20/best-practices-in-epub-cover-images/
+        """
+        if not self.images_are_extracted():
+            raise RuntimeError('extract_covers cannot be run before extract_images')
+
         # pad out the filename to four digits
-        cover_file = "{tmp}/{item_bookpath}_jp2/{item_bookpath}_{num:0>4}.jp2".format(
-            tmp=self.tmpdir, item_bookpath=self.item_bookpath, num=cover_leaf
-        )
-        try:
-            with ZipFile(self.jp2_zip) as f:
-                f.extractall(self.tmpdir)
-        except BadZipFile as e:
-            self.logger.error(
-                "extraction problem with {}".format(self.jp2_zip)
-            )
-            raise BadZipFile
+        cover_jp2 = "{tmp}/{item_bookpath}_jp2/{item_bookpath}_{num:0>4}.jp2".format(
+            tmp=self.tmpdir, item_bookpath=self.item_bookpath, num=self.get_cover_leaf())
+        cover_bmp = '{}/cover.bmp'.format(self.tmpdir)
+        cover_png = '{}/cover.png'.format(self.tmpdir)
+        self.logger.debug("jp2: %s & bmp: %s & png: %s" % (
+            cover_jp2, cover_bmp, cover_png))
+
         # convert the JP2K file into a usable format for the cover
-        f, e = os.path.splitext(os.path.basename(cover_file))
         imageobj = ImageFactory(self.image_processor)
         try:
-            imageobj.crop_image(cover_file, self.cover_img)
+            imageobj.crop_image(cover_jp2, cover_bmp)  # jp2 to bmp
+            imageobj.convert_bmp2png(
+                # bmp seems to break in ADE and other readers;
+                # convert bmp to png
+                cover_bmp, cover_png, resize=(800, 1200))
         except RuntimeError as e:
             # for failed image creation, keep processing the epub
             self.logger.error(e)
 
+        self.book.set_cover(
+            'images/cover.png', open(cover_png, 'rb').read())
+        cover = self.book.items[-1]
+        self.logger.debug(cover)
```

Comment:
QUESTION: Is this debug code intended to remain?

Reply:
This is only run when --debug is provided. It's how there logger is configured